speech/captionasync/captionasync.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2016 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Command captionasync sends an audio data to the Google Speech API
+// and pulls the operation status and the transcript.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+
+	speech "cloud.google.com/go/speech/apiv1beta1"
+	speechpb "google.golang.org/genproto/googleapis/cloud/speech/v1beta1"
+	longrunningpb "google.golang.org/genproto/googleapis/longrunning"
+)
+
+const usage = `Usage: captionasync <audiofile>
+
+Audio file is required to be 16-bit signed little-endian encoded
+with a sample rate of 16000.
+`
+
+func main() {
+	flag.Parse()
+	if len(os.Args) < 2 {
+		fmt.Fprintln(os.Stderr, usage)
+		os.Exit(2)
+	}
+
+	ctx := context.Background()
+	client, err := speech.NewClient(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	opName, err := send(client, os.Args[1])
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	resp, err := wait(client, opName)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Print the results.
+	for _, result := range resp.Results {
+		for _, alt := range result.Alternatives {
+			fmt.Printf("\"%v\" (confidence=%3f)\n", alt.Transcript, alt.Confidence)
+		}
+	}
+}
+
+func send(client *speech.Client, filename string) (string, error) {
+	ctx := context.Background()
```

Comment:
This should take a context as first arg. It doesn't really matter in a small program like this, but people will be looking to these for good style, and we should be encouraging passing of contexts. They are also likely to cut and paste this function for use in their own programs, where it may be important to pass the context.